src/rnpkeys/rnpkeys.cpp

```diff
@@ -253,7 +265,17 @@ rnp_cmd(rnp_cfg_t *cfg, cli_rnp_t *rnp, optdefs_t cmd, const char *f)
         return import_keys(cfg, rnp, f);
     case CMD_GENERATE_KEY: {
         if (f == NULL) {
-            f = rnp_cfg_getstr(cfg, CFG_USERID);
+            list ids = NULL;
+            if (rnp_cfg_copylist_str(cfg, &ids, CFG_USERID) && list_length(ids) > 0) {
+                if (list_length(ids) == 1) {
+                    f = (fs = (char*)list_front(ids)).c_str();
+                } else {
+                    fprintf(stderr, "only userid is supported for generated keys\n");
+                    list_destroy(&ids);
+                    return false;
+                }
```

Comment:
As the previous - helper function + `list_length(rnp_cfg_get_list())` to check list's length.